--- origsrc/mate-session-manager-1.2.0/configure.ac	2012-02-29 15:03:09.000000000 -0600
+++ src/mate-session-manager-1.2.0/configure.ac	2012-04-17 02:26:59.260736700 -0500
@@ -69,9 +69,17 @@ PKG_CHECK_MODULES(MATE_SESSION,
         gio-2.0 >= $GLIB_REQUIRED
         gtk+-$GTK_API_VERSION >= $GTK_REQUIRED
         dbus-glib-1 >= $DBUS_GLIB_REQUIRED
-        upower-glib >= $UPOWER_REQUIRED
 )
 
+PKG_CHECK_MODULES(UPOWER,
+        upower-glib >= $UPOWER_REQUIRED,
+        have_upower=true,
+        have_upower=false)
+
+if test "$have_upower" = true; then
+  AC_DEFINE(HAVE_UPOWER, 1, [Set to 1 if upower-glib is available.])
+fi
+
 PKG_CHECK_MODULES(SESSION_PROPERTIES,
         glib-2.0 >= $GLIB_REQUIRED
         gtk+-$GTK_API_VERSION >= $GTK_REQUIRED
--- origsrc/mate-session-manager-1.2.0/mate-session/gsm-logout-dialog.c	2012-02-29 15:03:09.000000000 -0600
+++ src/mate-session-manager-1.2.0/mate-session/gsm-logout-dialog.c	2012-04-17 02:26:59.280737900 -0500
@@ -27,7 +27,9 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 
+#ifdef HAVE_UPOWER
 #include <upower.h>
+#endif
 
 #include "gsm-logout-dialog.h"
 #include "gsm-consolekit.h"
@@ -50,7 +52,9 @@ struct _GsmLogoutDialogPrivate
 {
         GsmDialogLogoutType  type;
 
+#ifdef HAVE_UPOWER
         UpClient            *up_client;
+#endif
 #ifdef HAVE_SYSTEMD
         GsmSystemd          *systemd;
 #endif
@@ -142,7 +146,9 @@ gsm_logout_dialog_init (GsmLogoutDialog
         gtk_window_set_keep_above (GTK_WINDOW (logout_dialog), TRUE);
         gtk_window_stick (GTK_WINDOW (logout_dialog));
 
+#ifdef HAVE_UPOWER
         logout_dialog->priv->up_client = up_client_new ();
+#endif
 
         logout_dialog->priv->consolekit = gsm_get_consolekit ();
 
@@ -166,10 +172,12 @@ gsm_logout_dialog_destroy (GsmLogoutDial
                 logout_dialog->priv->timeout_id = 0;
         }
 
+#ifdef HAVE_UPOWER
         if (logout_dialog->priv->up_client) {
                 g_object_unref (logout_dialog->priv->up_client);
                 logout_dialog->priv->up_client = NULL;
         }
+#endif
 
         if (logout_dialog->priv->consolekit) {
                 g_object_unref (logout_dialog->priv->consolekit);
@@ -182,13 +190,21 @@ gsm_logout_dialog_destroy (GsmLogoutDial
 static gboolean
 gsm_logout_supports_system_suspend (GsmLogoutDialog *logout_dialog)
 {
+#ifdef HAVE_UPOWER
         return up_client_get_can_suspend (logout_dialog->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
 gsm_logout_supports_system_hibernate (GsmLogoutDialog *logout_dialog)
 {
+#ifdef HAVE_UPOWER
         return up_client_get_can_hibernate (logout_dialog->priv->up_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
--- origsrc/mate-session-manager-1.2.0/mate-session/gsm-manager.c	2012-02-29 15:03:09.000000000 -0600
+++ src/mate-session-manager-1.2.0/mate-session/gsm-manager.c	2012-04-17 02:26:59.284738100 -0500
@@ -38,7 +38,9 @@
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
+#ifdef HAVE_UPOWER
 #include <upower.h>
+#endif
 
 #include <gtk/gtk.h> /* for logout dialog */
 #include <mateconf/mateconf-client.h>
@@ -137,8 +139,10 @@ struct GsmManagerPrivate
         DBusGProxy             *bus_proxy;
         DBusGConnection        *connection;
 
+#ifdef HAVE_UPOWER
         /* Interface with other parts of the system */
         UpClient               *up_client;
+#endif
 };
 
 enum {
@@ -1006,6 +1010,7 @@ manager_perhaps_lock (GsmManager *manage
 static void
 manager_attempt_hibernate (GsmManager *manager)
 {
+#ifdef HAVE_UPOWER
         gboolean  can_hibernate;
         GError   *error;
         gboolean  ret;
@@ -1024,11 +1029,13 @@ manager_attempt_hibernate (GsmManager *m
                         g_error_free (error);
                 }
         }
+#endif
 }
 
 static void
 manager_attempt_suspend (GsmManager *manager)
 {
+#ifdef HAVE_UPOWER
         gboolean  can_suspend;
         GError   *error;
         gboolean  ret;
@@ -1047,6 +1054,7 @@ manager_attempt_suspend (GsmManager *man
                         g_error_free (error);
                 }
         }
+#endif
 }
 
 static void
@@ -2219,10 +2227,12 @@ gsm_manager_dispose (GObject *object)
                 manager->priv->mateconf_client = NULL;
         }
 
+#ifdef HAVE_UPOWER
         if (manager->priv->up_client != NULL) {
                 g_object_unref (manager->priv->up_client);
                 manager->priv->up_client = NULL;
         }
+#endif
 
         G_OBJECT_CLASS (gsm_manager_parent_class)->dispose (object);
 }
@@ -2452,7 +2462,9 @@ gsm_manager_init (GsmManager *manager)
                           G_CALLBACK (on_presence_status_changed),
                           manager);
 
+#ifdef HAVE_UPOWER
         manager->priv->up_client = up_client_new ();
+#endif
 
         /* MateConf setup */
         mateconf_client_add_dir (manager->priv->mateconf_client,
@@ -2976,10 +2988,15 @@ gsm_manager_can_shutdown (GsmManager *ma
         gboolean can_suspend;
         gboolean can_hibernate;
 
+#ifdef HAVE_UPOWER
         g_object_get (manager->priv->up_client,
                       "can-suspend", &can_suspend,
                       "can-hibernate", &can_hibernate,
                       NULL);
+#else
+        can_suspend = FALSE;
+        can_hibernate = FALSE;
+#endif
 
         g_debug ("GsmManager: CanShutdown called");
 
