https://github.com/mate-desktop/mate-session-manager/issues/86

--- origsrc/mate-session-manager-1.8.1/data/mate.desktop.in	2014-03-05 14:39:13.000000000 -0600
+++ src/mate-session-manager-1.8.1/data/mate.desktop.in	2015-06-18 18:38:53.538699500 -0500
@@ -5,4 +5,5 @@ Exec=mate-session
 TryExec=mate-session
 Icon=
 Type=Application
-Keywords=launch;MATE;desktop;session;
\ No newline at end of file
+Keywords=launch;MATE;desktop;session;
+DesktopNames=MATE
